### PR TITLE
IOS-294 Initial SPM integration set-up

### PR DESCRIPTION
**Pull request by ettore**
**What's this do?**
Changes to integrate NYPLUtilities and OverdriveProcessor as Swift Packages.

**Why are we doing this? (w/ JIRA link if applicable)**
https://jira.nypl.org/browse/IOS-294

**How should this be tested? / Do these changes have associated tests?**
This is not meant to create any visible changes for the user, so a regression will cover this.

**Dependencies for merging? Releasing to production?**
This PR depends on the following other PRs, which I will need to merge first:
https://github.com/NYPL-Simplified/iOS-Utilities/pull/2
https://github.com/NYPL-Simplified/NYPLAudiobookToolkit/pull/157
https://github.com/NYPL-Simplified/audiobook-ios-overdrive/pull/11

After I merge those, I'll also need to update the cartfile in NYPLAEToolkit, and after than update cartfile, SPM and submodule references here on Simplified.

**Does this include changes that require a new SimplyE/Open eBooks build for QA?**
yes

**Has the application documentation been updated for these changes?**
yes

**Did someone actually run this code to verify it works?**
@ettore

### CODE REVIEW

File: Simplified.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -6081,6 +6044,63 @@
 			defaultConfigurationName = Release;
 		};
 /* End XCConfigurationList section */
+
+/* Begin XCRemoteSwiftPackageReference section */
+		73579044276BE7F2009F1ADF /* XCRemoteSwiftPackageReference "iOS-Utilities" */ = {
+			isa = XCRemoteSwiftPackageReference;
+			repositoryURL = "https://github.com/NYPL-Simplified/iOS-Utilities";
+			requirement = {
+				branch = "task/IOS-294/utilities";
```
Comment (ErnestFan) [inline Simplified.xcodeproj/project.pbxproj:6053]: Do we need to update this before merging? Same for the Overdrive framework reference below

Comment (ettore) [reply Simplified.xcodeproj/project.pbxproj:6053]: oh yeah, definitely... this PR is not ready yet, but I wanted to get a head start to see if I was gonna get build errors... and of course I did get some 🤦‍♂️

Comment (ErnestFan) [reply Simplified.xcodeproj/project.pbxproj:6050]: Oh sure, let me know when it's ready.

Comment (ettore) [reply Simplified.xcodeproj/project.pbxproj]: this is now ready, although I will still need to update the commit refs once I merge the dependent PRs.

File: Simplified.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -842,7 +842,6 @@
 		73A794C72549098700C59CC1 /* NYPLOPDSAcquisitionPathEntryMinimal.xml in Resources */ = {isa = PBXBuildFile; fileRef = 73A794C62549095700C59CC1 /* NYPLOPDSAcquisitionPathEntryMinimal.xml */; };
 		73A794CA25492C9800C59CC1 /* NYPLFake.swift in Sources */ = {isa = PBXBuildFile; fileRef = 73A794C925492C9800C59CC1 /* NYPLFake.swift */; };
 		73A794CB25492C9800C59CC1 /* NYPLFake.swift in Sources */ = {isa = PBXBuildFile; fileRef = 73A794C925492C9800C59CC1 /* NYPLFake.swift */; };
-		73A7BA4E25A62FEA00C2906C /* R2LCPClient.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 73A7BA4D25A62FE900C2906C /* R2LCPClient.framework */; };
```
Comment (ErnestFan) [inline Simplified.xcodeproj/project.pbxproj:845]: Is this not needed anymore?

**Review by ErnestFan — APPROVED** [review]

**Comment by ettore** [comment]
the last force push is to adjust the commit references in the dependencies, no other changes

File: Simplified.xcodeproj/project.pbxproj (hunk 2 above)
Comment (ettore) [reply Simplified.xcodeproj/project.pbxproj:845]: it is for SimplyE, but I believe that's just in the test target, where it's not needed

**Pull request merged**